html/user-activation/activation-same-and-cross-origin.sub.html

```diff
@@ -0,0 +1,148 @@
+    }
+
+    promise_test(async (t) => {
+      soChild.src = "./resources/child-one.html";
+      xoChild.src =
+        "http://{{hosts[alt][]}}:{{ports[http][0]}}/html/user-activation/resources/child-two.html";
+      const [, unclickedCrossOrigin] = await Promise.all([
+        receiveMessage("child-one-loaded"),
+        receiveMessage("child-two-loaded"),
+      ]);
+      const soActivation = soChild.contentWindow.navigator.userActivation;
+      assert_false(
+        navigator.userActivation.isActive,
+        "navigator.userActivation.isActive must be false"
+      );
+      assert_false(
+        navigator.userActivation.hasBeenActive,
+        "navigator.userActivation.hasBeenActive must be false"
+      );
+
+      assert_false(soActivation.isActive, "child-one isActive must be false");
+      assert_false(
+        soActivation.hasBeenActive,
+        "child-one hasBeenActive must be false"
+      );
+      assert_false(
+        unclickedCrossOrigin.isActive,
+        "child-two isActive must be false"
+      );
+      assert_false(
+        unclickedCrossOrigin.hasBeenActive,
+        "child-two hasBeenActive must be false"
+      );
+    }, "Check Initial states of user activation are all false");
+
+    promise_test(async (t) => {
+      await test_driver.click(soChild);
+      const xoActivation = await requestXOReport();
+      const soActivation = soChild.contentWindow.navigator.userActivation;
+      assert_true(
+        navigator.userActivation.isActive,
+        "navigator.userActivation.isActive must be true"
+      );
+      assert_true(
+        navigator.userActivation.hasBeenActive,
+        "navigator.userActivation.hasBeenActive must be true"
+      );
+      assert_true(soActivation.isActive, "child-one isActive must be true");
+      assert_true(
+        soActivation.hasBeenActive,
+        "child-one hasBeenActive must be true"
+      );
+      assert_false(xoActivation.isActive, "child-two isActive must be false");
+      assert_false(
+        xoActivation.hasBeenActive,
+        "child-two hasBeenActive must be false"
+      );
+    }, "Check that activating a same-origin navigable doesn't activate a cross origin navigable");
+
+    promise_test(async (t) => {
+      await consumeTransientActivation();
+      const soActivation = soChild.contentWindow.navigator.userActivation;
+      assert_false(
+        navigator.userActivation.isActive,
+        "navigator.userActivation.isActive must be false"
+      );
+      assert_true(
+        navigator.userActivation.hasBeenActive,
+        "navigator.userActivation.hasBeenActive must be true"
+      );
+      assert_false(soActivation.isActive, "child-one isActive must be false");
+      assert_true(
+        soActivation.hasBeenActive,
+        "child-one hasBeenActive must be true"
+      );
+      const [, xoActivation] = await Promise.all([
+        test_driver.click(xoChild),
+        receiveMessage("child-two-clicked"),
+      ]);
+      assert_false(
```

Comment:
@mustaqahmed, sorry to bother you... can you sanity check something for me here? Above:

1. I've consumed the user activation of the window.
2. I click on the *cross-origin* iframe.

Now, if I'm reading HTML correctly, the parent frame *should not* have transient activation because it's not same origin as iframe. 

Is that your understanding too?

Reply:
Never mind... it turns out it should.